PocketKit/Sources/PocketKit/Tags/TagsFilterViewModel.swift

```diff
@@ -60,22 +59,24 @@ class TagsFilterViewModel: ObservableObject {
     }
 
     func delete(tags: [String]) {
-        let event = SnowplowEngagement(type: .general, value: nil)
-        let contexts: Context = UIContext.button(identifier: .tagsDelete)
-        tracker.track(event: event, [contexts])
+        trackTagsDelete()
         tags.forEach { tag in
             guard let tag: Tag = fetchedTags.filter({ $0.name == tag }).first else { return }
             source.deleteTag(tag: tag)
         }
     }
 
-    func rename(from oldName: String, to newName: String) {
-        let event = SnowplowEngagement(type: .general, value: nil)
-        let contexts: Context = UIContext.button(identifier: .tagsSaveChanges)
-        tracker.track(event: event, [contexts])
-        guard let tag: Tag = fetchedTags.filter({ $0.name == oldName }).first else { return }
+    func rename(from oldName: String?, to newName: String, completion: () -> Void) {
+        guard let oldName else {
+            Log.capture(message: "Unable to rename tag")
+            return
+        }
+        let newName = newName.lowercased()
+        guard let tag: Tag = fetchedTags.filter({ $0.name == oldName }).first,
+              !fetchedTags.compactMap({ $0.name }).contains(newName) else { return }
         source.renameTag(from: tag, to: newName)
-        refreshView = true
+        trackTagRename()
+        completion()
```

Comment:
open to suggestions if a better way to wait until this function completes before performing the scrolling logic

Reply:
@cyndichin how come we cant wait for the function to just return? Am i missing something?

Reply:
hmm, originally I wanted to make sure that `.onChange(of: renameTagText)` was only being called if we rename the tag. But since we're adding the completion block in both cases, then it doesn't matter anymore actually. I can remove them.